src/ChangeStream.php

```diff
@@ -98,15 +96,13 @@ public function next()
     {
         $resumable = false;
         try {
-            if ($this->valid()) {
-                $this->previousValid = true;
-            }
             $this->csIt->next();
             if ($this->valid()) {
-                $this->resumeToken = $this->extractResumeToken($this->csIt->current());
-                if ($this->previousValid) {
+                if ($this->hasAdvanced) {
                     $this->key++;
                 }
+                $this->hasAdvanced = true;
+                $this->resumeToken = $this->extractResumeToken($this->csIt->current());
```

Comment:
I'd propose keeping this line at the top of the conditional scope for the PHPLIB-327 commit. A second commit for PHPLIB-335 can then move it to the end of the conditional scope.

If you're adding regression tests for each issue in the respective commits, the second commit for PHPLIB-335 may also need to adjust PHPLIB-327's test.